src/ldp/http/SimpleBodyParser.ts

```diff
@@ -0,0 +1,52 @@
+import { BodyParser } from './BodyParser';
+import { HttpRequest } from '../../server/HttpRequest';
+import { Quad } from 'rdf-js';
+import { QuadRepresentation } from '../representation/QuadRepresentation';
+import { RepresentationMetadata } from '../representation/RepresentationMetadata';
+import { StreamParser } from 'n3';
+import { TypedReadable } from '../../util/TypedReadable';
+import { UnsupportedMediaTypeHttpError } from '../../util/errors/UnsupportedMediaTypeHttpError';
+import 'jest-rdf';
+
+export class SimpleBodyParser extends BodyParser {
+  private static readonly contentTypes = [
+    'application/n-quads',
+    'application/trig',
+    'application/n-triples',
+    'text/turtle',
+    'text/n3',
+  ];
+
+  public async canHandle(input: HttpRequest): Promise<void> {
+    const contentType = input.headers['content-type'];
+
+    if (contentType && !SimpleBodyParser.contentTypes.some((type): boolean => contentType.includes(type))) {
+      throw new UnsupportedMediaTypeHttpError('This parser only supports RDF data.');
+    }
+  }
+
+  public async handle(input: HttpRequest): Promise<QuadRepresentation> {
+    const contentType = input.headers['content-type'];
+
+    if (!contentType) {
+      return undefined;
+    }
+
+    const specificType = contentType.split(';')[0];
+
+    const metadata: RepresentationMetadata = {
+      raw: [],
+      profiles: [],
+      contentType: specificType,
+    };
+
+    // StreamParser is a Readable but typings are incorrect at time of writing
+    const quads: TypedReadable<Quad> = input.pipe(new StreamParser()) as unknown as TypedReadable<Quad>;
```

Comment:
If we do these nasty casts a lot, we may want to consider a helper function for it.

Reply:
I would hope this would be the only one (or not even if the typings get fixed). Should we ever need this when referencing our own internal classes it means we did something wrong with the typings.